src/js/helpers/cleanData.js

```diff
@@ -49,15 +49,31 @@ export const cleanLinks = (links, pathPrefix) => {
   if (!links || !links.edges) return null;
 
   return links.edges.map(({ node: link }) => {
-    const { title, slug, ...rest } = link;
+    const { title, text, slug, ...rest } = link;
     return {
       url: `${pathPrefix || ''}/${slug}`,
-      text: title,
+      text: title || text,
       ...rest,
     };
   });
 };
 
+export const cleanNavigation = navigation => {
+  const { allThemes } = navigation;
+
+  if (!allThemes || !allThemes.edges) return null;
+
+  let cleanedNavigation = cleanLinks(allThemes, '/themes');
+  cleanedNavigation.map(theme => {
```

Comment:
don't you need to set whatever the map returns to a new var so when you return it on line 74 its the updated array? mapping won't mutate `cleanedNavigation`. Or wait, the mutations you're doing are inside the object so they do persist? hmm, I'm probably wrong but maybe double check?

Reply:
yep, the second half of what you're saying is what's going on -- doing mutations on the object directly and since its memory reference sticks, the change persists when the function exits.